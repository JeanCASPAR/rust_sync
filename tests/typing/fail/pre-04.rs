use rust_sync::sync;

sync! {
    node oui() = ()
    where
        x : bool = false -> y,
        y : bool = pre (!x);
}

fn main() {}
