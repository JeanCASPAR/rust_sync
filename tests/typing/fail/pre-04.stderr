error: this expression will cause an access to an undefined value
 --> tests/typing/fail/pre-04.rs:9:20
  |
9 |         y : bool = pre (!x);
  |                    ^^^^^^^^
